devito/petsc/initialize.py

```diff
@@ -12,10 +12,21 @@
 _petsc_initialized = False
 
 
-def PetscInitialize():
+global _petsc_clargs
+
+
+def PetscInitialize(clargs=sys.argv):
     global _petsc_initialized
+    global _petsc_clargs
```

Comment:
You define the `_petsc_clargs` twice here. I think the one outside gets declared every time you import from this file right?

Reply:
Yes, but importantly not set. Globals are nasty, but this is largely to work around a piece of broken PETSc functionality IIRC